src/CMakeLists.txt

```diff
@@ -159,6 +159,13 @@ else()
   target_compile_definitions(cellml PUBLIC ${LIBXML2_DEFINITIONS})
 endif()
 
+# Prevent superfluous warnings from ranlib about libcellml.a having no symbols
+# on macOS.
+if(APPLE)
+  set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
+  set(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
```

Comment:
Would it be better to do this through a target property on STATIC_LIBRARY_OPTIONS?

Reply:
What do you have in mind exactly? Happy for you to suggest some code. :)

Reply:
```
set_property(
    TARGET cellml
    APPEND
    PROPERTY STATIC_LIBRARY_OPTIONS "-no_warning_for_no_symbols"
)
```

Reply:
I have just given it a try, but to no avail:

```
FAILED: src/libcellml.a
: && /usr/local/Cellar/cmake/3.16.3/bin/cmake -E remove src/libcellml.a && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar qc src/libcellml.a -no_warning_for_no_symbols src/CMakeFiles/cellml.dir/component.cpp.o src/CMakeFiles/cellml.dir/componententity.cpp.o src/CMakeFiles/cellml.dir/debug.cpp.o src/CMakeFiles/cellml.dir/entity.cpp.o src/CMakeFiles/cellml.dir/error.cpp.o src/CMakeFiles/cellml.dir/generator.cpp.o src/CMakeFiles/cellml.dir/generatorprofile.cpp.o src/CMakeFiles/cellml.dir/importedentity.cpp.o src/CMakeFiles/cellml.dir/importsource.cpp.o src/CMakeFiles/cellml.dir/logger.cpp.o src/CMakeFiles/cellml.dir/model.cpp.o src/CMakeFiles/cellml.dir/namedentity.cpp.o src/CMakeFiles/cellml.dir/orderedentity.cpp.o src/CMakeFiles/cellml.dir/parser.cpp.o src/CMakeFiles/cellml.dir/printer.cpp.o src/CMakeFiles/cellml.dir/reset.cpp.o src/CMakeFiles/cellml.dir/units.cpp.o src/CMakeFiles/cellml.dir/utilities.cpp.o src/CMakeFiles/cellml.dir/validator.cpp.o src/CMakeFiles/cellml.dir/variable.cpp.o src/CMakeFiles/cellml.dir/version.cpp.o src/CMakeFiles/cellml.dir/xmlattribute.cpp.o src/CMakeFiles/cellml.dir/xmldoc.cpp.o src/CMakeFiles/cellml.dir/xmlnode.cpp.o src/CMakeFiles/cellml.dir/xmlutils.cpp.o && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib src/libcellml.a && /usr/local/Cellar/cmake/3.16.3/bin/cmake -E touch src/libcellml.a && :
ar: -no_warning_for_no_symbols: No such file or directory
ninja: build stopped: subcommand failed.
```
I then tried to replace `STATIC_LIBRARY_OPTIONS` with `STATIC_LIBRARY_FLAGS`, as suggested [here](https://stackoverflow.com/a/52124651), but still no luck. The same when replacing `APPEND` with `APPEND_STRING` with either `STATIC_LIBRARY_OPTIONS` or `STATIC_LIBRARY_FLAGS`.

Reply:
After consideration I think it would be better to exclude the `debug.cpp` file from the build instead.  And then add in an option to the configuration to enable debug helper methods.